protos/flyteidl/admin/execution.proto

```diff
@@ -289,6 +289,10 @@ message ExecutionSpec {
 
     // Controls how to select an available cluster on which this execution should run.
     ClusterAssignment cluster_assignment = 20;
+
+    // Allows for a workflow to be flagged as interruptible for a single execution.
+    // Omitting this field uses the workflow's defined value as a default.
+    bool interruptible = 21;
```

Comment:
since proto3 doesn't have a concept of required fields, the default value for bools will always be false. things get weird when the workflow sets interruptible to true but we explicitly want to disable interruptible for a single execution

can we use one-ofs here like we do with notifications to make setting this to false explicit (and not default): https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/admin/execution.proto#L256

Reply:
That's something I considered - the current implementation does not allow for an override to _disable_ interruptible for a single execution. I initially tried specing it with this in mind, however I believe using `oneof` does not work here due to the [`WorkflowExecutionConfigInterface`](https://github.com/flyteorg/flyteadmin/pull/410/files#diff-9b311b663f43eccf79220bfcae3a565ad26d83532c945f39cfc1f7ba029c22e3R451) used in flyteadmin - I did not find a way to get the generated `oneof` methods to play along nicely with the interface.

I did however attempt using the predefined [`google.protobuf.BoolValue`](https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/bool-value) in my [initial commit](https://github.com/flyteorg/flyteidl/pull/287/commits/c6be9b432163984bce2594d3f4925aee1a9f69c4#diff-724a7e8bb673471da21d8cb4a8ef634198bc10cc7b61965a920673c2b7408a54), however switched to using regular bools for simplicity. Using the `BoolValue` wrapper did produce compatible code though, so that could be used.
Alternatively, I could extract the interruptible override from flyteadmin's `mergeIntoExecConfig` so this incompatibility would not arise.

From our internal teams, I've heard that explicity _disabling_ was not a requirement, as they'd rather just enable it, but if that's something you'd be interested in, I can take another look.

Reply:
I think with the `WorkflowExecutionConfigInterface` you can change the method return type accordingly to the generated go struct for the oneof or alternate field instead, right? and determine the execution workflowExecConfig.Interruptible bit accordingly

in any case, cool! I didn't know about BoolValue but that seems like a nillable option we can use instead

also, sorry I'm not quite following on what you mean by

> Alternatively, I could extract the interruptible override from flyteadmin's mergeIntoExecConfig so this incompatibility would not arise and oneof could be used - I believe that would also get rid of the currently unused interruptible flag in the LaunchPlanSpec.

I think for the purpose of completeness and since this is an open source feature it would be nice to have a means of disabling as well

Reply:
> I think with the `WorkflowExecutionConfigInterface` you can change the method return type accordingly to the generated go struct for the oneof or alternate field instead, right? and determine the execution workflowExecConfig.Interruptible bit accordingly

Yeah, but unfortunately the methods generated for the `oneof` of the `ExecutionSpec` and `LaunchPlanSpec` are not equal and not assignable to one another, at least from what I've seen. In any case, `BoolValue` works for all cases and solves the issue, so I'll go with that 🙂 

> also, sorry I'm not quite following on what you mean by
> 
> > Alternatively, I could extract the interruptible override from flyteadmin's mergeIntoExecConfig so this incompatibility would not arise and oneof could be used - I believe that would also get rid of the currently unused interruptible flag in the LaunchPlanSpec.

Instead of setting the interruptible flag in the `mergeIntoExecConfig` function, I could've done it at a different place, which wouldn't require the `WorkflowExecutionConfigInterface` to have an `GetInterruptible` method, but for the sake of consistency (and since `BoolValue` solves the issue anyways), I'll keep it there.

> I think for the purpose of completeness and since this is an open source feature it would be nice to have a means of disabling as well

I figured as much, changes are nearly ready and will be pushed tomorrow morning.

Reply:
Changes using `BoolValue` have been pushed